### PR TITLE
melody96: Added layout for non-hotswap PCB

### PR DESCRIPTION
**Pull request by tpstevens**
## Description

[According to YMDK (see Manuals section)](https://ymdkey.com/products/kit-1-aluminum-cnc-top-acrylic-cnc-bottom-ymdk-melody-96-hotswap-pcb-keyboard-kit), using the hotswap layout on the non-hotswap PCB requires swapping the position of ENTER and \|. Many users aren't aware of this:
* [Reddit post 9 months ago](https://old.reddit.com/r/MechanicalKeyboards/comments/i3qzio/help_flashed_keymap_with_qmk_and_now_enter_and/)
* [Reddit post 6 months ago](https://old.reddit.com/r/MechanicalKeyboards/comments/ja4yf5/hi_building_my_second_mech_with_the_ymd96_and/)
* [Github issue open since 2019](https://github.com/qmk/qmk_firmware/issues/6471)
* [Github issue open since 12 days ago](https://github.com/qmk/qmk_firmware/issues/12666)

To fix it, I created a new layout macro for the default ANSI hotswap keymap that supports the non-hotswap PCB.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #6471
* #12666

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
I think this layout macro needs a more accurate name, as the non-hotswap board supports more than one layout (hence why the hotswap version gets its own macro). It appears to be the same as the existing `LAYOUT_std60_split_num0` but with two extra keys (the split right shift and 5 keys to the right of space vs 4).

**Comment by tpstevens** [comment]
Fair enough. How about LAYOUT_non_hotswap_c to match YMDK's documentation? https://ymdkey.com/products/kit-1-aluminum-cnc-top-acrylic-cnc-bottom-ymdk-melody-96-hotswap-pcb-keyboard-kit

**Comment by fauxpark** [comment]
Since there is only one possible hotswap layout, all others can be assumed to be for the soldered version, so there is no need to mention it at all. This may be a little outside the scope of this PR, but I'd say the curent layouts (plus this new one) should be named as such:
```
LAYOUT_all
LAYOUT_hotswap
LAYOUT_ansi_split_num0
LAYOUT_ansi_split_rshift_num0
```
And of course this list is not exhaustive as the board can support many more combinations of layout options, as shown by the diagram below the Layout A/B/C/D images.

**Comment by tpstevens** [comment]
That's a good point about removing "hotswap" from the name. I double-checked the configurator, and the default LAYOUT layout does work with the non-hotswap PCB with most of the same key positions I proposed in this PR. The only real difference is that I have a KC_NO in the key to the right of the split left shift.

I think that users reported the issues above because they weren't sure which parameters in the LAYOUT macro corresponded to which keys (e.g. if I have the full-size left-shift instead of the split shift, does KC_LSFT go in the far left key or one to the right?) and thought that LAYOUT_hotswap would be a convenient way to get that layout for their non-hotswap PCB. That's how I ran into the issue, at least. 

With that in mind, I think that ideally we should document in the readme (which appears in the Configurator) which keys in the LAYOUT macro correspond to which configurations in YMDK's non-hotswap diagram. That way we don't need to create new macros for new layouts. However, we've already added one macro for LAYOUT_std60_split_num0 - do you think it's worth allowing contributors to add more? Adding better documentation to the default LAYOUT already makes it easier to understand, but I think it would be nice to keep LAYOUT_std60_split_num0 and get more presets.

I like the naming convention you propose, but I would suggest using YMDK's A, B, C, and D where possible:

````
LAYOUT_all (currently called LAYOUT)
LAYOUT_ansi_split_num0 (currently called LAYOUT_std60_split_num0)
LAYOUT_c (what I proposed in this PR)
LAYOUT_hotswap_c (currently called LAYOUT_hotswap)
````

Adding the rest of YMDK's layouts in both hotswap and non-hotswap forms would also be nice. I'm happy to do the work - I created my own hotswap layout C with mill-max sockets and can test all the combinations pretty easily.

**Comment by fauxpark** [comment]
The YMDK names could be aliases of more specific layout names (the ones I proposed). See for example the Whitefox's layout macros:
https://github.com/qmk/qmk_firmware/blob/master/keyboards/whitefox/whitefox.h
You can see we prefer a more "standardised" naming convention. This is to help with the Community Layouts feature, where all boards that support eg. an ANSI TKL layout would have a layout macro named `LAYOUT_tkl_ansi` and one can simply write a keymap using that layout and compile+flash it on any of said boards.

I also realised that although it is not set up, the Melody96 would support the [`96_ansi`](https://github.com/qmk/qmk_firmware/tree/master/layouts/default/96_ansi) layout - it is exactly the same as `LAYOUT_hotswap` - as well as `96_iso`. Since this new non-hotswap layout is the same but swaps two matrix positions, I think this problem should be solved by creating a new subfolder specifically for the hotswap board, which contains only a single `LAYOUT_96_ansi` layout.

**Comment by tpstevens** [comment]
I was unaware of Community Layouts; that's a neat idea. So ideally we would have a new hotswap folder with 96_iso, 96_ansi, and two other layouts to cover all 4 options?

**Comment by fauxpark** [comment]
As far as I'm aware there would only be one layout for the hotswap folder, unless they are making multiple hotswap boards for each layout.

**Comment by tpstevens** [comment]
I don't know if they actually use multiple PCB variants for each hotswap type, but I do know that the buyer can only choose one for their board and can't switch to another layout. That's why I figured we should include multiple layouts (to avoid the same user confusion for the LAYOUT macro as far as which keycode goes where).

**Comment by tpstevens** [comment]
@fauxpark I'm having a little trouble with moving the hotswap variant of the PCB to its own folder. I created a hotswap folder with a new info.json, rules.mk, and hotswap.h/.c with LAYOUT_all/LAYOUT_b/LAYOUT_96_ansi. I then created a new 96_ansi keymap and moved the existing hotswap zunger keymap to hotswap/keymaps. Both keymaps fail to compile (e.g. qmk compile -kb melody96/hotswap -km 96_ansi) -- QMK isn't seeing the new layout macros.

Am I messing something up with the folder structure or include paths? My second commit (which I'll revert) has the changes.

File: keyboards/melody96/hotswap/hotswap.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "hotswap.h"
```
Comment (drashna) [inline keyboards/melody96/hotswap/hotswap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/melody96/hotswap/hotswap.h [hunk 2]
```diff
@@ -0,0 +1,108 @@
+#pragma once
```
Comment (drashna) [inline keyboards/melody96/hotswap/hotswap.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/melody96/hotswap/hotswap.h [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	{ K060,  K061,  K062,  K063,  K064,  KC_NO, K066,  KC_NO, KC_NO }, \
+	{ KC_NO, K071,  K072,  K073,  K074,  K075,  K076,  K077,  K078  }, \
+	{ K080,  K081,  K082,  KC_NO, K084,  K085,  K086,  K087,  K088  }, \
+	{ KC_NO, K091,  K092,  K093,  K094,  K095,  K096,  K097,  K098  }, \
+	{ K100,  K101,  K102,  KC_NO, K104,  K105,  K106,  K107,  K108  }, \
+	{ K110,  K111,  K112,  K113,  K114,  K115,  K116,  K117,  K118  }  \
+}
+
+#define LAYOUT LAYOUT_all
+#define LAYOUT_c LAYOUT_96_ansi
```
Comment (drashna) [inline keyboards/melody96/hotswap/hotswap.h:108]:
```suggestion
```

File: keyboards/melody96/hotswap/keymaps/96_ansi/keymap.c [hunk 4]
```diff
@@ -0,0 +1,74 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/melody96/hotswap/keymaps/96_ansi/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/melody96/hotswap/hotswap.h (hunk 3 above)
Comment (tpstevens) [reply keyboards/melody96/hotswap/hotswap.h:108]: Thanks for the feedback -- I'll add the license headers. What's the rationale for removing the LAYOUT aliases? I was following the style of [whitefox](https://github.com/qmk/qmk_firmware/blob/master/keyboards/whitefox/whitefox.h) based on fauxpark's suggestions.

This is still a work in progress, as I need a little bit of guidance as far as the compilation errors I'm having with the hotswap layout. Not sure of the folder structure yet.

**Pull request closed without merging**